build.sbt

```diff
@@ -1,31 +1,62 @@
-crossScalaVersions := Seq("2.13.6", "2.12.14")
-scalaVersion := crossScalaVersions.value.head
+import xerial.sbt.Sonatype._
 
-name := "library-example"
+ThisBuild / scalaVersion := "3.0.1-RC2"
 
-// also used as a `groupId` by Sonatype
-organization := "ch.epfl.scala"
+lazy val library = project
+  .in(file("library"))
+  .settings(
+    name := "library-example",
+    organization := "ch.epfl.scala", // also used as a `groupId` by Sonatype
+    description := "A library that does nothing useful",
+    crossScalaVersions := Seq("3.0.0", "2.13.6", "2.12.14"),
+    scalaVersion := crossScalaVersions.value.head,
+    libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test,
+    testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+  ).settings(publishSettings)
 
-libraryDependencies += "com.github.scalaprops" %% "scalaprops" % "0.8.3" % Test
-testFrameworks += new TestFramework("scalaprops.ScalapropsFramework")
+lazy val publishSettings = Def.settings(
+  sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch")),
+  // indicate the open source licenses that apply to our project
+  licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt")),
+  // publish to the Sonatype repository
+  publishTo := sonatypePublishToBundle.value,
+  // binary compatibility check
+  mimaPreviousArtifacts := Set.empty // Disabled on `master` branch
+)
 
-description := "A library that does nothing useful"
+lazy val documentation = project
+  .in(file("site"))
+  .settings(siteSettings)
+  .enablePlugins(MdocPlugin)
+  .dependsOn(library)
 
-import xerial.sbt.Sonatype._
-sonatypeProjectHosting := Some(GitHubHosting("scalacenter", "library-example", "julien.richard-foy@epfl.ch"))
-// indicate the open source licenses that apply to our project
-licenses := Seq("APL2" -> url("http://www.apache.org/licenses/LICENSE-2.0.txt"))
-// publish to the Sonatype repository
-publishTo := sonatypePublishToBundle.value
+lazy val siteSettings = {
+  /*
+  Combine scaladoc and mdoc.
+  To do so,
+   1. copy site to target/processed_site
+   2. use mdoc to process site/docs/ **.md
+   3. mdoc outputs new md files in target/processed_site/docs
+   4. scaladoc runs and outputs files in site-ouput
+  */
+  val tmpSiteDir = "processed_site"
+  val outputSiteDir = "site-output"
```

Comment:
I would like to reiterate my previous comment: would it be worth to integrate with sbt-site and to be seen as a website generator from the point of view of sbt-site? (the benefits that come to my mind is that all the existing projects that are built on top of sbt-site would work out of the box, also we would get the task `previewSite` for free–but maybe there is already an equivalent in scaladoc?)

Reply:
I am not aware of any preview feature in scaladoc.
I did not find any documentation in the `sbt-site` website about writing a custom site generator, maybe we can discuss this idea with the project maintainers

Reply:
So, after internal discussion, we concluded that this could be left for a separate PR.